### PR TITLE
fix(worker-utils): Use default import for "child_process"

### PR DESCRIPTION
**Pull request by manzt**
Further discussion here: https://github.com/hms-dbmi/viv/pull/480#issuecomment-892970913

TL;DR - when `@loaders.gl/worker-utils` [transitioned to TS](visgl/loaders.gl@787271a#diff-653aabc2ef3b70398fe5a382c9f7c6359a5ecdd223901c6b15ea54b6ce418e67), changing 

```javascript
import ChildProcess from 'child_process';
```

to

```typescript
import {spawn, ChildProcess, SpawnOptionsWithoutStdio} from 'child_process';
```

broke our build when we tried to upgrade to deck.gl 8.5. This is because the `"empty"` resolution for some bundlers for node builtins is only a `default` export, so using named imports won't be resolved. This may seem like a bundler/bundler-plugin issue, but bundlers will resolve `package.json` "browser" differently. Using a `default` import for node-builtins will likely reduce friction with using loaders.gl with different bundlers in the future.

### CODE REVIEW

**Comment by manzt** [comment]
Example with `rollup`: https://gist.github.com/d690cf5f9360a5184bb1a2a763aff6f7

```
// index.js
import * as utils from '@loaders.gl/worker-utils';
console.log(utils);
```

```
// rollup.config.js
import resolve from '@rollup/plugin-node-resolve';

export default {
	input: 'index.js',
	output: { format: 'esm', file: 'bundle.js' },
	plugins: [
		resolve({ browser: true }),
	]
}
```


```bash
rollup -c
# [!] Error: 'spawn' is not exported by node-resolve:empty.js, imported by node_modules/@loaders.gl/worker-utils/dist/esm/lib/process-utils/child-process-proxy.js 
# https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module
#
# node_modules/@loaders.gl/worker-utils/dist/esm/lib/process-utils/child-process-proxy.js (2:9)
# 1: import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
# 2: import { spawn } from 'child_process';
            ^
```

File: modules/worker-utils/src/lib/process-utils/child-process-proxy.ts [hunk 1]
```diff
@@ -39,7 +40,7 @@ const DEFAULT_PROPS: ChildProcessProxyProps = {
 export default class ChildProcessProxy {
   id: string;
   props: ChildProcessProxyProps = {...DEFAULT_PROPS};
-  private childProcess: ChildProcess | null = null;
+  private childProcess: ReturnType<typeof ChildProcess['spawn']> | null = null;
```
Comment (ibgreen) [inline modules/worker-utils/src/lib/process-utils/child-process-proxy.ts:43]: If `import type` works, then we can drop this. If it does not work, can we define `type ChildProcess = ...` at the top of the file, with a comment explaining why we are doing this?

**Review by ibgreen — APPROVED** [review]

**Comment by ibgreen** [comment]
> . This is because the "empty" resolution for some bundlers for node builtins is only a default export, so using named imports won't be resolved

We are using `import * as fs` in other places to solve a similar problem - in which case we get an empty object. Seems more natural than importing default.

**Comment by manzt** [comment]
> We are using `import * as fs` in other places to solve a similar problem - in which case we get an empty object. Seems more natural than importing default.

That's a great suggestion! Thanks.

File: modules/worker-utils/src/lib/process-utils/child-process-proxy.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /* eslint-disable no-console */
-import {spawn, ChildProcess, SpawnOptionsWithoutStdio} from 'child_process';
+import ChildProcess from 'child_process';
```
[the remark this answers is not in the document]
Comment (manzt) [reply modules/worker-utils/src/lib/process-utils/child-process-proxy.ts]: To avoid naming conflicts, I used the `import * as` syntax and then accessed types from the module object:

```typescript
import * as ChildProcess from 'child_process';

export type ChildProcessProxyProps = {
  /* ... */
  spawn?: ChildProcess.SpawnOptionsWithoutStdio;
  /* ... */
};

export default class ChildProcessProxy {
  /* ... */
  private childProcess: ChildProcess.ChildProcess | null = null;
  /* ... */
}
```
